src/libraries/System.Diagnostics.Process/tests/ProcessWaitingTests.cs

```diff
@@ -169,6 +348,44 @@ public void WaitForPeerProcess()
             Assert.Equal(RemoteExecutor.SuccessExitCode, child2.ExitCode);
         }
 
+        [Fact]
+        public async Task WaitAsyncForPeerProcess()
+        {
+            using Process child1 = CreateProcessLong();
+            child1.Start();
+
+            using Process child2 = CreateProcess(async peerId =>
+            {
+                Process peer = Process.GetProcessById(int.Parse(peerId));
+                Console.WriteLine("Signal");
+                using (var cts = new CancellationTokenSource(WaitInMS))
+                {
+                    await peer.WaitForExitAsync(cts.Token);
```

Comment:
I don't see any tests where WaitForExitAsync is used without a token on a process that exits successfully / normally.  Are there any?

Reply:
`SingleProcess_WaitAsyncAfterExited` waits without a token on a process that has already exited, but I hadn't written any tests that have a potentially unbounded wait in order to avoid possibly blocking CI.

Now that you've pointed it out though I've added another test `WaitAsyncForProcess` that doesn't pass a cancellation token, but still doesn't wait unboundedly.